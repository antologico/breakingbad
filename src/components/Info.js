import React from 'react'
import PropTypes from 'prop-types'

import { FormattedDate, FormattedMessage } from 'react-intl'
import { CharacterWrapper, CharacterSection, CharacterInfoList, CharacterImage, HorizontalRule, LinkButton, Title } from '../ui'

const Info = ({
    name,
    image,
    nickname,
    appearance,
    occupation,
    birthday,
    status,
    portrayed,
}) => <CharacterWrapper>
        <Title data-test='name'>
            {name} <br /> <sup><i>"{nickname}"</i></sup>
        </Title>
        <CharacterImage data-test='image' src={image} alt={name} height='300px' width='100%'/>
        <CharacterSection>
            <FormattedMessage defaultMessage='Birthday' id='birthday' />
            : <strong>{birthday !== 'Unknown'
                ? <FormattedDate data-test='birthday'
                    value={new Date(birthday)}
                    year="numeric"
                    month="long"
                    day="numeric"
                    weekday="long"
                />
                : <FormattedMessage defaultMessage='-' id='unknown' />}</strong>
        </CharacterSection>
        <CharacterSection>
            <FormattedMessage defaultMessage='Status' id='status' />
            : <strong>{status}</strong>
        </CharacterSection>
        <CharacterSection>
            <FormattedMessage defaultMessage='Actor' id='actor' />
            : <strong>{portrayed}</strong>
        </CharacterSection>
        <CharacterSection data-test='occupation'>
            <FormattedMessage defaultMessage='Occupation' id='occupation' />:
            {occupation && <CharacterInfoList>
                {occupation.map(item => (<li key={item}>{item}</li>))}
            </CharacterInfoList>}
        </CharacterSection>
        <CharacterSection data-test='seassons'>
            <FormattedMessage defaultMessage='Seassons' id='seassons' />:
            {appearance && <CharacterInfoList>
                {appearance.map(item => (<li key={item}>{item}</li>))} 
            </CharacterInfoList>}
        </CharacterSection>
        <LinkButton to='/' data-test='back-button'>
            <span><FormattedMessage defaultMessage='Return' id='return' /></span>
            <span>&#8617;</span> 
        </LinkButton>
    </CharacterWrapper>

Info.propTypes =Â {
    name: PropTypes.string,
    nickname: PropTypes.string,
    image: PropTypes.string,
    status: PropTypes.string,
    portrayed: PropTypes.string,
    appearance: PropTypes.arrayOf(PropTypes.string),
    occupation: PropTypes.arrayOf(PropTypes.string),
}

export default Info