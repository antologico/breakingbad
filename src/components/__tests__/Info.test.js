import React from 'react'
import { create } from '../../tests/utils'
import Info from '../Info'

jest.mock('react-router-dom', () => ({
  Link: ({children}) => <a id='link'>{children}</a>
}))

describe(' component', () => {
  test('renders properly', () => {
    const component = create(
        <Info 
          name='Walt'
          image='image.png'
          nickname='Nickname'
          appearance={['1', '2']}
          occupation={['Bank']}
          birthday={['2020-02-02']}
          status='single'
          portrayed='Actor Name'
        />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})