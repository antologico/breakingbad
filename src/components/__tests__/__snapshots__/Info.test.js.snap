// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` component renders properly 1`] = `
<article
  className="sc-iBkjds bINJJu"
>
  <h1
    className="sc-bczRLJ ibOPpr"
    data-test="name"
  >
    Walt
     
    <br />
     
    <sup>
      <i>
        "
        Nickname
        "
      </i>
    </sup>
  </h1>
  <img
    alt="Walt"
    className="sc-eCYdqJ dJkwBv"
    data-test="image"
    height="300px"
    src="image.png"
    width="100%"
  />
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Fecha de nacimiento
    : 
    <strong>
      domingo, 2 de febrero de 2020
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Estado
    : 
    <strong>
      single
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Actor/Actriz
    : 
    <strong>
      Actor Name
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
    data-test="occupation"
  >
    Ocupación
    :
    <ul
      className="sc-ftvSup kDPIQF"
    >
      <li>
        Bank
      </li>
    </ul>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
    data-test="seassons"
  >
    Temporadas
    :
    <ul
      className="sc-ftvSup kDPIQF"
    >
      <li>
        1
      </li>
      <li>
        2
      </li>
    </ul>
  </section>
  <a
    id="link"
  >
    <span>
      Volver
    </span>
    <span>
      ↩
    </span>
  </a>
</article>
`;

exports[` component renders properly without date 1`] = `
<article
  className="sc-iBkjds bINJJu"
>
  <h1
    className="sc-bczRLJ ibOPpr"
    data-test="name"
  >
    Walt
     
    <br />
     
    <sup>
      <i>
        "
        Nickname
        "
      </i>
    </sup>
  </h1>
  <img
    alt="Walt"
    className="sc-eCYdqJ dJkwBv"
    data-test="image"
    height="300px"
    src="image.png"
    width="100%"
  />
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Fecha de nacimiento
    : 
    <strong>
      Desconocido
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Estado
    : 
    <strong>
      single
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
  >
    Actor/Actriz
    : 
    <strong>
      Actor Name
    </strong>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
    data-test="occupation"
  >
    Ocupación
    :
    <ul
      className="sc-ftvSup kDPIQF"
    >
      <li>
        Bank
      </li>
    </ul>
  </section>
  <section
    className="sc-hKMtZM gZOnDg"
    data-test="seassons"
  >
    Temporadas
    :
    <ul
      className="sc-ftvSup kDPIQF"
    >
      <li>
        1
      </li>
      <li>
        2
      </li>
    </ul>
  </section>
  <a
    id="link"
  >
    <span>
      Volver
    </span>
    <span>
      ↩
    </span>
  </a>
</article>
`;
